### PR TITLE
bulk data upload added for criterias

### PR DESCRIPTION
**Pull request by aman-tunerlabs**

### CODE REVIEW

File: controllers/v1/criteriasController.js [hunk 1]
```diff
@@ -934,5 +934,40 @@ module.exports = class Criterias extends Abstract {
...
+        }
+
+        let criteriaData = await csv().fromString(req.files.criterias.data.toString())
+
+        Promise.all(criteriaData.map(async criteria => {
+          let findQuery = {
+            externalId: criteria.criteriaID
+          }
+          let updateQuery = { $set: { "name": criteria.criteriaName, "rubric.levels.L1.description": criteria.L1, "rubric.levels.L2.description": criteria.L2, "rubric.levels.L3.description": criteria.L3, "rubric.levels.L4.description": criteria.L4 } }
+          criteria = await database.models.criterias.findOneAndUpdate(
```
Comment (aks30) [inline controllers/v1/criteriasController.js]: Will this create if it is not there ?

Comment (aman-tunerlabs) [reply controllers/v1/criteriasController.js]: Changed findOneAndUpdate to create.

File: controllers/v1/criteriasController.js [hunk 2]
```diff
@@ -934,5 +934,144 @@ module.exports = class Criterias extends Abstract {
...
+              "0125747659358699520",
+              "0125748495625912324"
+            ],
+            evidences: [],
+            deleted: false,
+            externalId: criteria.criteriaID,
+            owner: req.userDetails.id,
+            timesUsed: 12,
+            weightage: 20,
+            aremarks: "",
```
Comment (aks30) [inline controllers/v1/criteriasController.js]: Please remove aremarks, it should be remarks! @aman-tunerlabs

File: controllers/v1/criteriasController.js [hunk 3]
```diff
@@ -934,5 +934,144 @@ module.exports = class Criterias extends Abstract {
...
+            name: criteria.criteriaName,
+            description: criteria.criteriaName,
+            criteriaType: "auto",
+            score: "",
+            flag: "",
+            rubric: {
```
Comment (aks30) [inline controllers/v1/criteriasController.js]: Please make this dynamic, it should work with how many ever levels are there in CSV. @aman-tunerlabs

File: controllers/v1/criteriasController.js [hunk 4]
```diff
@@ -934,5 +934,144 @@ module.exports = class Criterias extends Abstract {
...
+              name: criteria.criteriaName,
+              description: criteria.criteriaName,
+              type: "auto",
```
Comment (aks30) [inline controllers/v1/criteriasController.js]: Make criteria type also dynamic, accepted from CSV.

File: controllers/v1/criteriasController.js (hunk 2 above)
Comment (aman-tunerlabs) [reply controllers/v1/criteriasController.js]: changed

File: controllers/v1/criteriasController.js (hunk 3 above)
Comment (aman-tunerlabs) [reply controllers/v1/criteriasController.js]: modified

File: controllers/v1/criteriasController.js (hunk 4 above)
Comment (aman-tunerlabs) [reply controllers/v1/criteriasController.js]: modified

**Pull request merged**